extra_geom/tests/test_lpdmini_geometry.py

```diff
@@ -0,0 +1,200 @@
+    with pytest.raises(ValueError):
+        geom.rotate(np.ones((15, 3)))
+
+    # angles must be 3D
+    with pytest.raises(ValueError):
+        geom.rotate((1, 1))
+
+    # angles in radian
+    deg = geom.rotate((90, 0, 0), modules=np.s_[:1], tiles=np.s_[:1])
+    rad = geom.rotate((np.pi / 2, 0, 0), modules=np.s_[:1], tiles=np.s_[:1], degrees=False)
+    np.testing.assert_array_almost_equal(
+        deg.modules[0][1].corners(),
+        rad.modules[0][1].corners()
+    )
+
+
+def test_inspect():
+    geom = LPD_MiniGeometry.from_origin((11.4, 299))
+    # Smoketest
+    ax = geom.inspect()
+    assert isinstance(ax, Axes)
+
+
+def test_snap_assemble_data():
+    geom = LPD_MiniGeometry.from_origin()
+
+    stacked_data = np.zeros((1, 256, 256))
+    img, centre = geom.position_modules_fast(stacked_data)
+    assert img.shape == (284, 260)
+    assert tuple(centre) == (284, 260)
+    assert img[50, 50] == 0
+
+
+def test_to_distortion_array():
+    geom = LPD_MiniGeometry.from_origin()
+    # Smoketest
+    distortion = geom.to_distortion_array()
+    assert isinstance(distortion, np.ndarray)
+    assert distortion.shape == (256, 256, 4, 3)
+
+    # Coordinates in m, origin at corner; max x & y should be ~ 50cm
+    assert 0.12 < distortion[..., 1].max() < 0.70
+    assert 0.12 < distortion[..., 2].max() < 0.70
+    assert 0.0 <= distortion[..., 1].min() < 0.01
+    assert 0.0 <= distortion[..., 2].min() < 0.01
+
+
+def test_data_coords_to_positions():
+    geom = LPD_MiniGeometry.from_origin()
+
+    module_no = np.zeros(16, dtype=np.int16)
+    # Points near the centre of each tile
+    slow_scan = np.tile(np.linspace(16, 240, num=8, dtype=np.float32), 2)
+    fast_scan = np.array([64, 192], dtype=np.float32).repeat(8)
+
+    tileno, tile_ss, tile_fs = geom._module_coords_to_tile(slow_scan, fast_scan)
+    np.testing.assert_allclose(tileno,
+                       [7, 6, 5, 4, 3, 2, 1, 0, 8, 9, 10, 11, 12, 13, 14, 15])
+    np.testing.assert_allclose(tile_ss, 16)
+    np.testing.assert_allclose(tile_fs, 64)
+
+    res = geom.data_coords_to_positions(module_no, slow_scan, fast_scan)
+
+    assert res.shape == (16, 3)
+
+    resx, resy, resz = res.T
+
+    np.testing.assert_allclose(resz, 0)
+
+    assert (np.diff(resy[:8]) > 0).all()  # T1-T8 Monotonically increasing
+    assert (np.diff(resy[9:]) > 0).all()  # T9-T16 Monotonically increasing
+    assert -0.031 > resx.max() > resx.min() > -0.099
+
+
+def test_invert_xfel_lpd_geom(tmpdir):
+    src_file = pjoin(tests_dir, 'lpd_mar_18.h5')
+    dst_file = pjoin(str(tmpdir), 'lpd_inverted.h5')
+    invert_xfel_lpd_geom(src_file, dst_file)
+    with h5py.File(src_file, 'r') as fsrc, h5py.File(dst_file, 'r') as fdst:
+        np.testing.assert_array_equal(
+            fsrc['Q1/M1/Position'][:], -1 * fdst['Q1/M1/Position'][:]
+        )
+        np.testing.assert_array_equal(
+            fsrc['Q1/M1/T07/Position'][:], -1 * fdst['Q1/M1/T07/Position'][:]
+        )
```

Comment:
This is directly copied from the LPD 1M tests, right? Probably no need to repeat it.